api/lightwallet_rest.md

```diff
@@ -0,0 +1,402 @@
+| public_key   | bytes         | Bytes of output public key         |
+| rct          | bytes         | Bytes containing ringct commitment |
+
+> `global_index` is determined by the monero daemon. It is the offset from the
+> first time the amount appeared in the blockchain. After ringct, this is the
+> order of outputs as they appear in the blockchain.
+
+**random_outputs** object
+
+Randomly selected outputs for use in a ring signature.
+
+|   Field   |            Type          |       Description       |
+|-----------|--------------------------|-------------------------|
+|  amount   |       uint64-string      | XMR amount, 0 if ringct |
+| outputs * | array of random_output's | Selected outputs        |
+
+> `outputs` is omitted by the server if the `amount` does not have enough
+> mixable outputs.
+
+### Methods
+#### `get_address_info`
+Returns the minimal set of information needed to calculate a wallet balance.
+The server cannot calculate when a spend occurs without the spend key, so a
+list of candidate spends is returned.
+
+**Request** object
+
+|   Field   |      Type      |            Description                |
+|-----------|----------------|---------------------------------------|
+| address   | base58-address | Address to retrieve                   |
+| view_key  | binary         | View key bytes for authorization      |
+
+> If `address` is not authorized, the server must return a HTTP 405
+> "Forbidden" error.
+
+**Response** object
+
+|        Field         |       Type       |       Description         |
+|----------------------|------------------|---------------------------|
+| locked_funds         | uint64-string    | Sum of unspendable XMR    |
+| total_received       | uint64-string    | Sum of received XMR       |
+| total_sent           | uint64-string    | Sum of possibly spent XMR |
+| scanned_height       | uint64           | Current tx scan progress  |
+| scanned_block_height | uint64           | Current scan progress     |
+| start_height         | uint64           | Start height of response  |
+| transaction_height   | uint64           | Total txes sent in Monero |
+| blockchain_height    | uint64           | Current blockchain height |
+| spent_outputs        | array of spend's | Possible spend info       |
+| rates *              | rates            | Current exchange rates    |
+
+> `rates` is omitted if unavailable.
+
+#### `get_address_txs`
+Returns information needed to show transaction history. The server cannot
+calculate when a spend occurs without the spend key, so a list of candidate
+spends is returned.
+
+**Request** object
+
+|   Field  |       Type     |             Description               |
+|----------|----------------|---------------------------------------|
+| address  | base58-address | Address to retrieve                   |
+| view_key | binary         | View key bytes for authorization      |
+
+> If `address` is not authorized, the server must return a HTTP 405
```

Comment:
I think it could be more useful HTTP 403.

Details about HTTP 403 RFC 2616 :

> The server understood the request, but is refusing to fulfill it. Authorization will not help and the request SHOULD NOT be repeated. If the request method was not HEAD and the server wishes to make public why the request has not been fulfilled, it SHOULD describe the reason for the refusal in the entity. If the server does not wish to make this information available to the client, the status code 404 (Not Found) can be used instead.